library.c

```diff
@@ -1151,6 +1151,15 @@ PHP_REDIS_API int redis_type_response(INTERNAL_FUNCTION_PARAMETERS, RedisSock *r
     return SUCCESS;
 }
 
+PHP_REDIS_API int
+redis_config_response(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock, zval *z_tab, void *ctx) {
+    FailableResultCallback cb = ctx;
+
+    ZEND_ASSERT(cb == redis_boolean_response || cb == redis_mbulk_reply_zipped_raw);
```

Comment:
Why this assert here?

Reply:
Because I am assigning a function pointer from a `void*`, and I know the value should be one of those two callbacks.  
If the value is set incorrectly (or corrupted) it's pretty nasty undefined behavior.